examples/native_texture.rs

```diff
@@ -1,111 +1,160 @@
 #![cfg_attr(not(debug_assertions), windows_subsystem = "windows")] // hide console window on Windows in release
 
-use egui::{load::SizedTexture, ViewportId};
-use glium::{backend::glutin::SimpleWindowBuilder, glutin::surface::WindowSurface};
-use winit::event_loop::{EventLoop, EventLoopBuilder};
+use egui::{load::SizedTexture, TextureId, Vec2, ViewportId};
+use glium::{
+    backend::glutin::SimpleWindowBuilder, glutin::surface::WindowSurface, texture::SrgbTexture2d,
+};
+use std::rc::Rc;
+use winit::{
+    application::ApplicationHandler,
+    event::{StartCause, WindowEvent},
+    event_loop::{ActiveEventLoop, EventLoop},
+    window::{Window, WindowId},
+};
 
 fn main() {
-    let event_loop = EventLoopBuilder::with_user_event().build().unwrap();
-    let (window, display) = create_display(&event_loop);
-
-    let mut egui_glium =
-        egui_glium::EguiGlium::new(ViewportId::ROOT, &display, &window, &event_loop);
-
-    let png_data = include_bytes!("rust-logo-256x256.png");
-    let image = load_glium_image(png_data);
-    let image_size = egui::vec2(image.width as f32, image.height as f32);
-    // Load to gpu memory
-    let glium_texture = glium::texture::SrgbTexture2d::new(&display, image).unwrap();
-    // Allow us to share the texture with egui:
-    let glium_texture = std::rc::Rc::new(glium_texture);
-    // Allocate egui's texture id for GL texture
-    let texture_id = egui_glium
-        .painter
-        .register_native_texture(glium_texture, Default::default());
+    let event_loop = EventLoop::<()>::with_user_event().build().unwrap();
+
     // Setup button image size for reasonable image size for button container.
     let button_image_size = egui::vec2(32_f32, 32_f32);
 
-    let result = event_loop.run(move |event, target| {
+    let mut app = App {
+        graphics_context: None,
+        button_image_size,
+    };
+
+    let result = event_loop.run_app(&mut app);
+    result.unwrap()
+}
+
+struct App {
+    graphics_context: Option<GraphicsContext>,
```

Comment:
Is there a reason why this must be an `Option`? I would prefer to avoid the need for `if let` everywhere it is used.

Reply:
It was needed to be able to destroy and recreate the GL resources, but from my other comment, I see this may not be necessary. I'll try to rework it without the `Option` as I would definitely prefer to avoid it as well.